### PR TITLE
[0.12.2] backport: getblockchaininfo: make bip9_softforks an object, not an array.

### PR DESCRIPTION
**Pull request by maflcko**
This breaks the interface but it was tagged with 'Needs backport' and I think it is useful. Other opinions?

### CODE REVIEW

**Comment by laanwj** [comment]
Nah I think it makes sense to backport this in preparation for 0.13. Maybe mention in the release notes? Not sure how many people rely on this programmatically.

**Comment by MarcoFalke** [comment]
Updated the release notes just to be safe.

**Comment by MarcoFalke** [comment]
Anything holding this back?

**Comment by laanwj** [comment]
Nothing specific, just lack of review and testing as usual with backports.
I'll take a look at it.

File: doc/release-notes.md [hunk 1]
```diff
@@ -36,10 +36,13 @@ earlier.
 Notable changes
 ===============
 
-Example item
----------------------------------------
+Miscellaneous
```
Comment (laanwj) [inline doc/release-notes.md]: Nit: usually we call this category "RPC low-level API changes"
https://github.com/bitcoin/bitcoin/blob/v0.12.0/doc/release-notes.md#rpc-low-level-api-changes

**Comment by laanwj** [comment]
0.12 without this patch:
```
  "bip9_softforks": [
    {
      "id": "csv",
      "status": "locked_in"
    }
  ]
```
0.12 with this patch:
```json
  "bip9_softforks": {
    "csv": {
      "status": "started"
    }
  }
```
Master:
```json
  "bip9_softforks": {
    "csv": {
      "status": "active",
      "startTime": 1462060800,
      "timeout": 1493596800
    }
  }
```
ACK fa4327c

**Comment by luke-jr** [comment]
Eh, breaking API compatibility in a bugfix release? I don't see any real benefit to this. If people want the new API, they can just use 0.13. We've never broken API compatibility before in even a major release... now isn't the time to do it in a bugfix release. :/

**Comment by laanwj** [comment]
The idea would be to have the same API on 0.12 and 0.13, which makes e.g. testing easier.
But I don't care deeply whether this makes it into 0.12...

**Comment by laanwj** [comment]
Closing this, apparently it is controversial

**Pull request closed without merging**